strongbox-security/strongbox-authentication-providers/strongbox-ldap-authentication-provider/src/test/java/org/carlspring/strongbox/authentication/api/impl/ldap/LdapAuthenticationProviderTest.java

```diff
@@ -83,6 +83,24 @@ public void embeddedLdapServerRegistersExpectedAuthenticationProvider()
                 );
     }
 
+    @Test
+    public void base64EncodedPasswordsShouldWork()
+    {
+        UserDetails ldapUser = ldapUserDetailsService.loadUserByUsername("base64encoded-issue-1840");
+
+        assertThat(ldapUser).isInstanceOf(LdapUserDetailsImpl.class);
+        LdapUserDetails ldapUserDetails = (LdapUserDetailsImpl) ldapUser;
+
+        assertThat(ldapUserDetails.getDn()).isEqualTo("uid=base64encoded-issue-1840,ou=Users,dc=carlspring,dc=com");
+        assertThat(ldapUserDetails.getPassword()).isEqualTo("{bcrypt}JDJhJDEwJGxwd2x4eWp2WEt6TjFjY0NydzJQQnVaeC5lVmVzV2JmbVRic3JDYm9NVS5nc05XVmNaV01p");
+        assertThat(ldapUserDetails.getUsername()).isEqualTo("base64encoded-issue-1840");
+        assertThat(((List<SimpleGrantedAuthority>)ldapUser.getAuthorities()))
+                .contains(
+                        new SimpleGrantedAuthority(SystemRole.REPOSITORY_MANAGER.name()),
+                        new SimpleGrantedAuthority("USER_ROLE")
+                );
```

Comment:
This test actually doesn't do anything other than validating it can get a record from the LDAP server. :)
It should also have an assertion which checks if `passwordEncoder.matches(ldapUserDetails.getPassword(), "password")`.